src/rgw/rgw_iam_policy.h

```diff
@@ -446,6 +455,8 @@ struct Policy {
   Effect eval(const Environment& e,
 	      boost::optional<const rgw::auth::Identity&> ida,
 	      std::uint64_t action, const ARN& resource) const;
+
+  bool has_conditional(const string& conditional, bool partial=true) const;
```

Comment:
`partial=true` makes sense for use with `s3:ExistingObjectTag/`, but i tend to think that exact matches should be the default, with partial matches being opt-in. just thinking about new callers in the future using `has_conditional("foo")` without realizing that there's a `foobar` condition key also

Reply:
ack makes sense

Reply:
Not "required", but something to consider: what if you provided your default behavior, then an overload that took a predicated (which you would always call with the same parameters)?
template <typename MatchPred>
bool has_conditional(const std::string_view key_, MatchPred p) { return p(key, key_); } 
...and implement the other in terms of it.